apps/api/src/app/topics/e2e/filter-topics.e2e.ts

```diff
@@ -0,0 +1,103 @@
+    await createNewTopic(session, 'topic-key-2');
+    await createNewTopic(session, 'topic-key-3');
+  });
+
+  it('should return a validation error if the params provided are not in the right type', async () => {
+    const url = `${BASE_PATH}?page=first&pageSize=big`;
+    const response = await session.testAgent.get(url);
+
+    expect(response.statusCode).to.eql(400);
+    expect(response.body.error).to.eql('Bad Request');
+    expect(response.body.message).to.eql([
+      'page must be a number conforming to the specified constraints',
+      'pageSize must be a number conforming to the specified constraints',
+    ]);
+  });
+
+  it('should retrieve all the topics that exist in the database for the user if not query params provided', async () => {
+    const url = `${BASE_PATH}`;
+    const response = await session.testAgent.get(url);
+
+    expect(response.statusCode).to.eql(200);
+
+    const { data, totalCount, page, pageSize } = response.body;
+
+    expect(data.length).to.eql(3);
+    expect(totalCount).to.eql(3);
+    expect(page).to.eql(0);
+    expect(pageSize).to.eql(100);
+  });
+
+  it('should retrieve the topic filtered by the query param key for the user', async () => {
+    const topicKey = 'topic-key-2';
+    const url = `${BASE_PATH}?key=${topicKey}`;
+    const response = await session.testAgent.get(url);
+
+    expect(response.statusCode).to.eql(200);
+
+    const { data, totalCount, page, pageSize } = response.body;
+    const [topic] = data;
+
+    expect(data.length).to.eql(1);
+    expect(totalCount).to.eql(1);
+    expect(page).to.eql(0);
+    expect(pageSize).to.eql(100);
+    expect(topic._userId).to.eql(session.user._id);
+    expect(topic._environmentId).to.eql(session.environment._id);
+    expect(topic._organizationId).to.eql(session.organization._id);
+    expect(topic.key).to.eql(topicKey);
+    expect(topic.createdAt).to.exist;
+    expect(topic.updatedAt).to.exist;
+  });
+
+  it('should retrieve an empty response if filtering by a key that is not in the database for the user', async () => {
+    const topicKey = 'topic-key-not-existing';
+    const url = `${BASE_PATH}?key=${topicKey}`;
+    const response = await session.testAgent.get(url);
+
+    expect(response.statusCode).to.eql(200);
+
+    const { data, totalCount, page, pageSize } = response.body;
+
+    expect(data.length).to.eql(0);
+    expect(totalCount).to.eql(0);
+    expect(page).to.eql(0);
+    expect(pageSize).to.eql(100);
+  });
+
+  it('should ignore other query params and return all the topics that belong the user', async () => {
+    const url = `${BASE_PATH}?unsupportedParam=whatever`;
+    const response = await session.testAgent.get(url);
+
+    expect(response.statusCode).to.eql(200);
+
+    const { data, totalCount, page, pageSize } = response.body;
+
+    expect(data.length).to.eql(3);
+    expect(totalCount).to.eql(3);
+    expect(page).to.eql(0);
+    expect(pageSize).to.eql(100);
+  });
+});
```

Comment:
it would be nice to see a few tests with paginated data, for ex. the first and the second pages and using small page size

Reply:
also missing validation tests for the:
- negative page number
- negative page size
- page size > DEFAULT_TOPIC_LIMIT

Reply:
I replicated the pagination implemented in the activity feed so I expected it to be validated 😅 
I can add some of them though.